src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -201,13 +203,38 @@ public double getFee(LocalDateTime startPeriod, LocalDateTime endPeriod) {
     public double getFeePerStudent(Student student, LocalDateTime startPeriod, LocalDateTime endPeriod) {
         double fee = 0;
         for (Session session : student.getListOfSessions()) {
-            LocalDateTime dateTime = session.getSessionDate().getDateTime();
-            if (dateTime.compareTo(startPeriod) >= 0 && dateTime.compareTo(endPeriod) < 0) {
-                // This session date is within the period
-                fee += session.getFee().getFee();
+            if (session instanceof RecurringSession) {
+                fee += getRecurringSessionFee((RecurringSession) session, startPeriod, endPeriod);
+            } else {
+                fee += getSingleSessionFee(session, startPeriod, endPeriod);
             }
         }
         return fee;
     }
 
+    private double getRecurringSessionFee(RecurringSession recurringSession, LocalDateTime startPeriod,
+        LocalDateTime endPeriod) {
+        SessionDate startDate = new SessionDate(startPeriod);
+        // Minus one because the end date for numOfSessionBetween is inclusive
+        SessionDate endDate = new SessionDate(endPeriod.minusDays(1));
+        int numOfSession = recurringSession.numOfSessionBetween(startDate, endDate);
+
+        // Ensures that there is more than 0 session, so we can get the fees accordingly
+        if (numOfSession > 0) {
+            System.out.println("Fee: " + numOfSession);
+            return recurringSession.getFee().getFee() * numOfSession;
+        }
+        return 0;
```

Comment:
Do you want to throw an error if `numOfSession` is less than 1 instead, easier for debugging in the future.

Reply:
Haha ok! This code is moved to feeUtil, but then hearing ur advice, I have put an assertion that numOfSession >= 0 (Sometimes can be 0 depending on the period given). In this way, if we want to debug, we can just activate assertion, but then it wouldn't return an exception during normal activity.